keras_nlp/src/utils/transformers/safetensor_utils.py

```diff
@@ -43,10 +43,19 @@ def __init__(self, preset):
             self.safetensor_config = None
         self.safetensor_files = {}
 
+    def get_prefix(self, key, all_keys):
+        for k in all_keys:
+            if k.endswith(key) and k != key:
+                prefix = k[: -len(key)]
+                return prefix + key
+        return key
```

Comment:
I like the implementation! WDYT @mattdangerw

My 2 cents:
- We should use better variable naming.
- The name of the function is misleading.
- Adding a docstring here, so that we are well aware of the problem and how to solve it.

Reply:
Come to think of it. I think a better approach to this would be to loop over the keys all at once and have a one-to-one mapping of the hf keys and the keras keys.

Return the map, and then use that map later.

This bypasses the following:
- Looping over all the keys multiple times
- Using this function multiple times

Reply:
Yeah caching the prefix in some form sounds good.

Is the prefix always the same for all weights? If so, we could probably do something like this...

```python
def get_prefix(key, dict_like):
    if self.prefix is not None:
        return self.prefix
    keys = dict_like.keys()
    if key in keys:
        self.prefix = ""
        return self.prefix
    self.prefix = # Some code to figure out the correct prefix.
    return self.prefix
```

Reply:
No, the prefix are not same for all weights.

Taking an example of distilgpt2:

```
transformer.wte.weight
transformer.wpe.weight

transformer.h.0.attn.bias
transformer.h.0.attn.c_attn.bias
```